spec/operators/switchScan-spec.ts

```diff
@@ -0,0 +1,398 @@
+
+    expectObservable(result, unsub).toBe(expected);
+    expectSubscriptions(x.subscriptions).toBe(xsubs);
+    expectSubscriptions(y.subscriptions).toBe(ysubs);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+  });
+
+  it('should not break unsubscription chains when result is unsubscribed explicitly', () => {
+    const x =   cold(         '--a--b--c--d--e--|           ');
+    const xsubs =    '         ^         !                  ';
+    const y =   cold(                   '---f---g---h---i--|');
+    const ysubs =    '                   ^ !                ';
+    const e1 =   hot('---------x---------y---------|        ');
+    const e1subs =   '^                    !                ';
+    const expected = '-----------a--b--c----                ';
+    const unsub =    '                     !                ';
+
+    const observableLookup: Record<string, Observable<string>> = { x, y };
+
+    const result = e1.pipe(
+      mergeMap(x => of(x)),
+      switchScan((acc, value) => observableLookup[value], null),
+      mergeMap(x => of(x)),
+    );
+
+    expectObservable(result, unsub).toBe(expected);
+    expectSubscriptions(x.subscriptions).toBe(xsubs);
+    expectSubscriptions(y.subscriptions).toBe(ysubs);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+  });
+
+  it('should stop listening to a synchronous observable when unsubscribed', () => {
+    const sideEffects: number[] = [];
+    const synchronousObservable = concat(
+      defer(() => {
+        sideEffects.push(1);
+        return of(1);
+      }),
+      defer(() => {
+        sideEffects.push(2);
+        return of(2);
+      }),
+      defer(() => {
+        sideEffects.push(3);
+        return of(3);
+      })
+    );
+
+    of(null).pipe(
+      switchScan(() => synchronousObservable, null),
+      takeWhile((x) => x != 2) // unsubscribe at the second side-effect
+    ).subscribe(() => { /* noop */ });
+
+    expect(sideEffects).to.deep.equal([1, 2]);
+  });
+
+  it('should switch inner cold observables, inner never completes', () => {
+    const x =   cold(         '--a--b--c--d--e--|          ');
+    const xsubs =    '         ^         !                 ';
+    const y =   cold(                   '---f---g---h---i--');
+    const ysubs =    '                   ^                 ';
+    const e1 =   hot('---------x---------y---------|       ');
+    const e1subs =   '^                            !       ';
+    const expected = '-----------a--b--c----f---g---h---i--';
+
+    const observableLookup: Record<string, Observable<string>> = { x, y };
+
+    const result = e1.pipe(switchScan((acc, value) => observableLookup[value], null));
+
+    expectObservable(result).toBe(expected);
+    expectSubscriptions(x.subscriptions).toBe(xsubs);
+    expectSubscriptions(y.subscriptions).toBe(ysubs);
+    expectSubscriptions(e1.subscriptions).toBe(e1subs);
+  });
+
+  it('should handle a synchronous switch to the second inner observable', () => {
```

Comment:
thought: This looks like one of those tests that purports to test something that's synchronous, but isn't really synchronous-within-`subscribe`. Nothing to do here, this is just an observation related to the sync-within-`subscribe` marble syntax that we talked about.

Reply:
Yeah, now that I look at it, the test is overly complicated as well. I've simplified it a bit.